common/types/src/lib.rs

```diff
@@ -9,8 +9,10 @@ pub mod error;
 pub mod fees;
 pub mod gas;
 pub mod gateway;
+pub mod gateway_client_registration;
 pub mod helpers;
 pub mod mixnode;
 pub mod pending_events;
 pub mod transaction;
 pub mod vesting;
+pub mod wireguard;
```

Comment:
I wonder if we should place all the wireguard stuff in this crate behind the `wireguard` feature toggle? Then we don't have to worry about nym-connect and the wallet pulling in wireguard

Reply:
This is the `PeerPublicKey` type, so its neater this way, I'll put it behind a feature flag tough

Reply:
Practical problem was that there was a circular dependency between gateway and wireguard, so the types extracted it out